pkgs/applications/networking/syncthing/default.nix

```diff
@@ -22,7 +48,9 @@ let
 
       buildPhase = ''
         runHook preBuild
-        go run build.go -no-upgrade -version v${version} build ${target}
```

Comment:
the go compiler (just like clang) supports all the targets in a single binary. So we can set `GOOS="${pkgsBuildBuild.go.GOOS}" GOARCH="${pkgsBuildBuild.go.GOARCH}"` and everything just work.

Reply:
Can not confirm:


```
syncthing-armv7l-unknown-linux-gnueabihf> building
syncthing-armv7l-unknown-linux-gnueabihf> ++ export GOOS=linux GOARCH=amd64
syncthing-armv7l-unknown-linux-gnueabihf> ++ GOOS=linux
syncthing-armv7l-unknown-linux-gnueabihf> ++ GOARCH=amd64
syncthing-armv7l-unknown-linux-gnueabihf> ++ go build build.go
syncthing-armv7l-unknown-linux-gnueabihf> # runtime/cgo
syncthing-armv7l-unknown-linux-gnueabihf> armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command-line option '-m64'
```

It uses the wrong linker

Reply:
Try additionally setting `CC="${pkgsBuildBuild.stdenv.cc}/bin/cc`